### PR TITLE
[FIX] website_slides : page selector change embedded code

### PR DESCRIPTION
**Pull request by flvr-odoo**
Steps to reproduce:

  1. Install website_slides
  2. Create a new course, add a pdf
  3. view the pdf
  4. exit fullscreen
  5. click on "share" at the bottom
  6. change the start page

Issue:

  The embedded code does not change

Cause:

  The scope of the JQuery selector was inside the iframe.

Solution:

  Escape the ifram by using the top.document

Improvement:

  Adding a max and min on the input, there is no sense to select a page < 1 or above the amount of pages in the pdf

opw-2948880

### CODE REVIEW

**Review by hubvd — APPROVED** [review]

**Review by nle-odoo — APPROVED** [review]
I rebased on 14.0

it is not very clean to access the top window since I think that the embed could be in a third party website with a different domain that could have a cause of issue (if there is an error because we try to access another domain).

so I modified the solution so instead another code that was broken is fixed, but the max page is still not working.

a good fix would be for that IMO:

- sending an event in the embed when the page is loaded
- catching this event from the outside frame (so it would only be catched in Odoo)

or something else like checking the page number when we are changing the page number.

but here it seems to be an expected issue (there is a TODO that the max page is not working) so I think just doing the current change is enough without investing too much time.

**Comment by tde-banana-odoo** [comment]
@jeh-odoo @flch-odoo

File: addons/website_slides/static/src/js/slides.js [hunk 1]
```diff
@@ -115,8 +115,9 @@ publicWidget.registry.websiteSlidesEmbed = publicWidget.Widget.extend({
     _onIframeViewerReady: function (ev) {
         // TODO : make it work. For now, once the iframe is loaded, the value of #page_count is
         // still now set (the pdf is still loading)
-        var $iframe = $(ev.currentTarget);
-        var maxPage = $iframe.contents().find('#page_count').val();
+        var maxPage = parseInt(
+            $('iframe.o_wslides_iframe_viewer').contents().find('#page_count').text()
+        );
```
Comment (flch-odoo) [inline addons/website_slides/static/src/js/slides.js:120]: I think we could indeed move this to the onchange above, as it only matters when we actually want to change the page.
```js
if (!this.maxPage) {
   this.maxPage = parseInt(
        $('iframe.o_wslides_iframe_viewer').contents().find('#page_count').text()
    );
    // then perhaps set this max to the `<input>`?
}
```
We don't need more for a quick fix. (Keeping this handler with the todo for improving on it later if necessary,  but simplifying to `new SlideSocialEmbed(this, 0)`?)

**Review by flch-odoo — COMMENTED** [review]
Hello @nle-odoo , @flvr-odoo,
Can we try to fix it with something like this, since we're here?

cc @hubvd

**Comment by C3POdoo** [comment]
Dear @flvr-odoo,

Thank you for your contribution but the version 14.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**